### PR TITLE
Error Handling and Validation

### PR DESCRIPTION
**Pull request by joerivanarkel**

### CODE REVIEW

File: Common/Logger.cs [hunk 1]
```diff
@@ -23,9 +23,9 @@ public Logger(string Version, string Name = null)
     public void Log(string message, string orginFile, LogType logType)
     {
         var location = $"./bin/pirate{version}/logs/{logName}.log";
```
Comment (teusvanarkel) [inline Common/Logger.cs:25]: Move to highger level

File: Common/Logger.cs [hunk 2]
```diff
@@ -23,9 +23,9 @@ public Logger(string Version, string Name = null)
...
-        bool exists = System.IO.Directory.Exists($"./bin/pirate{version}/logs");
+        bool exists = Directory.Exists($"./bin/pirate{version}/logs");
         if (!exists)
-            System.IO.Directory.CreateDirectory($"./bin/pirate{version}/logs");
+            Directory.CreateDirectory($"./bin/pirate{version}/logs");
```
Comment (teusvanarkel) [inline Common/Logger.cs:29]: Move to higher level

File: Interpreter/Interpreters/BinaryOperationNodeInterpreter.cs [hunk 3]
```diff
@@ -10,7 +12,12 @@ public class BinaryOperationNodeInterpreter : BaseInterpreter
     public ILogger Logger { get; set; }
```
Comment (teusvanarkel) [inline Interpreter/Interpreters/BinaryOperationNodeInterpreter.cs:12]: Maybe logger in the base

Comment (teusvanarkel) [inline Interpreter/Interpreters/BinaryOperationNodeInterpreter.cs:12]: Make a private field ( starts with _ )

File: Interpreter/Interpreters/ValueNodeInterpreter.cs [hunk 4]
```diff
@@ -15,13 +16,23 @@ public class ValueNodeInterpreter : BaseInterpreter
     public ILogger Logger { get; set; }
     public ValueNodeInterpreter(INode node, ILogger logger)
     {
-        Node = node as IValueNode;
+        if (node is not IValueNode)
```
Comment (teusvanarkel) [inline Interpreter/Interpreters/ValueNodeInterpreter.cs:19]: Write on one line

File: Interpreter/Interpreters/ValueNodeInterpreter.cs [hunk 5]
```diff
@@ -15,13 +16,23 @@ public class ValueNodeInterpreter : BaseInterpreter
...
+
         Logger = logger;
         Logger.Log($"Created {this.GetType().Name} : \"{Node.ToString()}\"", this.GetType().Name, Common.Enum.LogType.INFO);
     }
 
     public override BaseValue VisitNode()
     {
+        if (Node.Value.Value == null)
+        {
+            Logger.Log($"{Node.Value.GetType().Name} does not contain a vaild value type.", this.GetType().Name, Common.Enum.LogType.ERROR);
```
Comment (teusvanarkel) [inline Interpreter/Interpreters/ValueNodeInterpreter.cs:33]: Remove log line and the rest on one line

File: Interpreter/Interpreters/ValueNodeInterpreter.cs [hunk 6]
```diff
@@ -35,8 +46,8 @@ public override BaseValue VisitNode()
             case TokenSyntax.IDENTIFIER:
                 return new Variable((string)Node.Value.Value, Logger);
         } 
-        Logger.Log($"{Node.Value.GetType().Name} does not contain a vaild value type.", this.GetType().Name, Common.Enum.LogType.ERROR);
-        return null;
+        Logger.Log($"{Node.Value.GetType().Name} is not trecognized as a BaseValue type.", this.GetType().Name, Common.Enum.LogType.ERROR);
```
Comment (teusvanarkel) [inline Interpreter/Interpreters/ValueNodeInterpreter.cs:49]: Remove log line

File: Shell/Commands/BuildCommand.cs [hunk 7]
```diff
@@ -8,30 +8,26 @@ namespace Shell.Commands
 {
     public class BuildCommand : Command
     {
-        public string version { get; set; }
-        public ILogger logger { get; set; }
         public ObjectSerializer ObjectSerializer { get; set;}
```
Comment (teusvanarkel) [inline Shell/Commands/BuildCommand.cs:11]: Make private

File: Shell/Commands/Command.cs [hunk 8]
```diff
@@ -4,8 +4,13 @@ namespace Shell.Commands;
 
 public abstract class Command
 {
-    public string version { get; set; }
-    public ILogger logger { get; set; }
+    public string Version { get; set; }
+    public ILogger Logger { get; set; }
```
Comment (teusvanarkel) [inline Shell/Commands/Command.cs:8]: Make protected field

File: Shell/Commands/Command.cs [hunk 9]
```diff
@@ -4,8 +4,13 @@ namespace Shell.Commands;
 
 public abstract class Command
 {
-    public string version { get; set; }
-    public ILogger logger { get; set; }
+    public string Version { get; set; }
```
Comment (teusvanarkel) [inline Shell/Commands/Command.cs:7]: Make private

**Review by teusvanarkel — COMMENTED** [review]
Done

**Comment by joerivanarkel** [comment]
**Summary**:
- Created Custom exceptions for Type Conversion and Parsing Exceptions
- Added validation for type conversions
- Replaced null return with errors
- Created base class constructors

**Pull request merged**